### PR TITLE
LttP, beemizer: support fine-tuned trap replacements

### PR DESCRIPTION
**Pull request by adampziegler**
Splits the single discrete 5-level `beemizer` LttP configuration into two separate weight blocks:
- `beemizer_total_chance` - the % chance a junk-fill item will be replaced
- `beemizer_trap_chance` - the % chance a replaced junk-fill item is a bee trap; all others are single bees

Following the pattern of weighted settings for `triforce_pieces_*`, custom percentages can be added, between 0 and 100, to the two weight blocks.

The comments from the original `beemizer` configuration block implied that the % of junk-fill items was predetermined across the entire set of junk items (with a % replaced as bee traps).  However, the beemizer item replacement logic is based on % _chance_ from two random rolls against _each_ junk-fill item:
- First, to see if the junk-fill item is replaced at all
- Then, to see if it should be a bee trap or a single bee

No functional changes have been made to the overall logic, aside from some short-circuits and a flattening of the `if` statement hierarchy.  The original "levels" have been preserved in the sample player YAML, now split across the two discrete configuration blocks.

~~I was unsure how to integrate the new settings with the webhost LttP settings page; however, it seems a number of recent and/or more complex options are missing from there, so I opted to leave them off, but still included in the sample YAML file for "expert" settings download, defaulting to 0% chance (off), as before.~~

### CODE REVIEW

**Comment by Berserker66** [comment]
Ah, well, you implemented the new options on the old system that we're trying to get rid of. Please use the Options module instead to declare your options, it's fine to rip out the old beemizer setting in doing so. Once you use the new system it will automatically add the new options to the template yamls and the website.

**Comment by adampziegler** [comment]
Ah, got it.  I had misinterpreted the difference in the sample and exported settings and didn't fully understand the conflict in options systems.

Should be fixed now.

File: worlds/alttp/Items.py [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
-                return "Bee" if isinstance(item, str) else world.create_item("Bee", player)
-        else:
-            return item
-    else:
+
+    if item_name not in trap_replaceable:
         return item
 
+    # first roll - replaceable item should be replaced, within beemizer_total_chance
+    if not world.beemizer_total_chance[player] or world.random.random() > (world.beemizer_total_chance[player] / 100):
```
Comment (Berserker66) [inline worlds/alttp/Items.py:10]: Why does this and the following need a `not world.beemizer_(total)_chance[player]` ?

**Comment by Berserker66** [comment]
Thank you!

**Pull request merged**